airflow/providers/microsoft/azure/hooks/asb.py

```diff
@@ -326,5 +342,32 @@ def receive_subscription_message(
                 max_message_count=max_message_count, max_wait_time=max_wait_time
             )
             for msg in received_msgs:
-                self.log.info(msg)
-                subscription_receiver.complete_message(msg)
+                self._process_message(msg, message_callback, subscription_receiver)
+
+    def _process_message(
+        self,
+        msg: ServiceBusReceivedMessage,
+        message_callback: MessageCallback | None,
+        receiver: ServiceBusReceiver,
+    ):
+        """
+        Process the message by calling the message_callback or logging the message.
+
+        :param msg: The message to process.
+        :param message_callback: Optional callback to process each message. If not provided, then
+            the message will be logged and completed. If provided, and throws an exception, the
+            message will be abandoned for future redelivery.
+        :param receiver: The receiver that received the message.
+        """
+        if message_callback is None:
+            self.log.info(msg)
```

Comment:
Is this right thing to print info for each callback? That looks pretty eccessive.

Reply:
That is the original code https://github.com/apache/airflow/blob/main/airflow/providers/microsoft/azure/hooks/asb.py#L292

Reply:
I see. Fine then :)